tests/lib/tools/log-analyzer

```diff
@@ -0,0 +1,511 @@
+    """
+
+    def __init__(self, verb, task, date, time, source_line):
+        self.type = ACTION
+        self.verb = verb
+        self.time = time
+        self.date = date
+        self.task = task
+        self.source_line = source_line
+
+    def __repr__(self):
+        return self.source_line
+
+    def __dict__(self):
+        return {
+            'type': 'action',
+            'date': self.date,
+            'time': self.time,
+            'verb': self.verb,
+            'task': self.task
+            }
+
+
+class Result:
+    """
+    Result represents the results for a spread run
+    The results can be: Successful, failed and aborted
+    """
+
+    def __init__(self, result_type, level, number, date, time, detail, source_line):
+        self.type = RESULT
+        self.result_type = result_type
+        self.level = level
+        self.number = number
+        self.time = time
+        self.date = date
+        self.detail = detail
+        self.source_line = source_line
+
+    def __repr__(self):
+        if self.detail:
+            return '{}{}'.format(self.source_line, str(self.detail))
+        return self.source_line
+
+    def __dict__(self):
+        prepared_detail = None
+        if self.detail:
+            prepared_detail = self.detail.__dict__()
+        return {
+            'type': self.type,
+            'date': self.date,
+            'time': self.time,
+            'result_type': self.result_type,
+            'level': self.level,
+            'number': self.number,
+            'detail': prepared_detail
+            }
+
+
+class Info:
+    """
+    Info represents the extra tasks information which is included in the spread log
+    The info can be: Error, Debug and Warning
+    """
+
+    def __init__(self, info_type, verb, task, extra, date, time, detail, source_line):
+        self.type = INFO
+        self.info_type = info_type
+        self.verb = verb
+        self.time = time
+        self.date = date
+        self.task = task
+        self.extra = extra
+        self.detail = detail
+        self.source_line = source_line
+
+    def __repr__(self):
+        if self.detail:
+            return '{}{}'.format(self.source_line, self.detail)
+        return self.source_line
+
+    def __dict__(self):
+        prepared_detail = None
+        if self.detail:
+            prepared_detail = self.detail.__dict__()
+        return {
+            'type': self.type,
+            'date': self.date,
+            'time': self.time,
+            'info_type': self.info_type,
+            'verb': self.verb,
+            'task': self.task,
+            'extra': self.extra,
+            'detail': prepared_detail
+            }
+
+
+class Detail:
+    """
+    Detail represents the extra information which are all the lines included for the info
+    """
+
+    def __init__(self, lines_limit, lines):
+        self.lines_limit = lines_limit
+        self.lines = lines
+
+    def _get_lines(self):
```

Comment:
I must say this is really weird. Is this type doing anything that a plain list is not doing?